objectRecognitionSource/src/main/java/io/vantiq/extsrc/objectRecognition/ObjectRecognitionCore.java

```diff
@@ -350,14 +353,17 @@ public void sendDataFromImage(ImageRetrieverResults imageResults, ExtensionServi
                }
                NeuralNetResults results = neuralNet.processImage(image, request);
                
-               // Send the normal message as the response if requested, otherwise just send the data 
-               if (request.get("sendFullResponse") instanceof Boolean && (Boolean) request.get("sendFullResponse")) {
-                   Map response = createMapFromResults(imageResults, results);
-                   client.sendQueryResponse(200, replyAddress, response);
-               } else if (results.getResults().isEmpty()) {
-                   client.sendQueryResponse(204, replyAddress, new LinkedHashMap<>());
-               } else {
-                   client.sendQueryResponse(200, replyAddress, results.getResults().toArray(new Map[0]));
+               // Don't send any data if results are null (for NoProcessor)
+               if (results != null) {
```

Comment:
Are there cases (_e.g._ failure) where results might be null?  Should we check for the NoProcessor processor rather than results == null?

If these are truly query responses, then doing nothing with processing == none is wrong.  We should still send a noData response in that case.  I think that the query in this case is used to manually poll the camera.  In that case, a query must still respond -- it'll just return no data having (presumably) saved the image as requested.  

We should definitely have a test for queries when the noProcessing option is selected.